### ISSUE TITLE
Internal compiler error involving boxing a closure...

### ISSUE THREAD

**Issue by Proksima** [rust-lang/rust#28397]
```
damien@fx-8150 ~/Documents/mkt $ cargo build
   Compiling mkt v0.1.0 (file:///home/damien/Documents/mkt)
error: internal compiler error: no type for node 1423: local stock (id=1423) in fcx 0x7ff9de3f1358
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
note: run with `RUST_BACKTRACE=1` for a backtrace
thread 'rustc' panicked at 'Box<Any>', src/libsyntax/diagnostic.rs:249


Could not compile `mkt`.
```

```
damien@fx-8150 ~/Documents/mkt $ export RUST_BACKTRACE=1; cargo build --verbose
       Fresh traitobject v0.0.1
       Fresh winapi v0.2.2
       Fresh rustc-serialize v0.3.16
       Fresh pkg-config v0.3.5
       Fresh unicase v1.0.0
       Fresh winapi-build v0.1.1
       Fresh bitflags v0.3.2
       Fresh typeable v0.1.2
       Fresh lazy_static v0.1.14
       Fresh libc v0.1.10
       Fresh httparse v0.1.6
       Fresh matches v0.1.2
       Fresh language-tags v0.0.7
       Fresh gcc v0.3.13
       Fresh log v0.3.2
       Fresh num_cpus v0.2.6
       Fresh url v0.2.37
       Fresh mime v0.1.0
       Fresh hpack v0.2.0
       Fresh advapi32-sys v0.1.2
       Fresh kernel32-sys v0.1.4
       Fresh solicit v0.4.3
       Fresh openssl-sys v0.6.5
       Fresh rand v0.3.11
       Fresh time v0.1.32
       Fresh openssl v0.6.5
       Fresh num v0.1.27
       Fresh cookie v0.1.21
       Fresh chrono v0.2.16
       Fresh hyper v0.6.11
   Compiling mkt v0.1.0 (file:///home/damien/Documents/mkt)
     Running `rustc src/main.rs --crate-name mkt --crate-type bin -g --out-dir /home/damien/Documents/mkt/target/debug --emit=dep-info,link -L dependency=/home/damien/Documents/mkt/target/debug -L dependency=/home/damien/Documents/mkt/target/debug/deps --extern hyper=/home/damien/Documents/mkt/target/debug/deps/libhyper-919a4d2425d9fde3.rlib --extern chrono=/home/damien/Documents/mkt/target/debug/deps/libchrono-0b07bc02e75d4749.rlib -L native=/usr/lib64 -L native=/home/damien/Documents/mkt/target/debug/build/openssl-sys-e0bd245bf09b1a3f/out`
error: internal compiler error: no type for node 1423: local stock (id=1423) in fcx 0x7f602fff1358
Could not compile `mkt`.

Caused by:
  Process didn't exit successfully: `rustc src/main.rs --crate-name mkt --crate-type bin -g --out-dir /home/damien/Documents/mkt/target/debug --emit=dep-info,link -L dependency=/home/damien/Documents/mkt/target/debug -L dependency=/home/damien/Documents/mkt/target/debug/deps --extern hyper=/home/damien/Documents/mkt/target/debug/deps/libhyper-919a4d2425d9fde3.rlib --extern chrono=/home/damien/Documents/mkt/target/debug/deps/libchrono-0b07bc02e75d4749.rlib -L native=/usr/lib64 -L native=/home/damien/Documents/mkt/target/debug/build/openssl-sys-e0bd245bf09b1a3f/out` (signal: 4)
```

The code that triggered this was:

```rust
struct StockFeature {
    identifier : String,
    routine    : Box<Fn(&market::Stock, &Option<usize>) -> market::Ratio>,
    period     : Option<usize>
}

fn parse_arguments(arguments: &Vec<String>) -> Vec<StockFeature> {
    const FEATURES: &'static [StockFeature] = &[
        StockFeature {
            identifier : "".to_string(),
            routine    : Box::new(|stock, period| price_to_yearly_range(stock)),
            period     : None
        }
    ];
}

fn price_to_yearly_range(stock: &market::Stock) -> market::Ratio { ... }
```

**Comment by arielb1** [comment]
Which version of rustc are you using?

**Comment by arielb1** [comment]
Fixed in 1.3: https://play.rust-lang.org/?gist=bf611611c5ab4c179e92&version=beta.

The error is correct (you can't allocate in a constant).
